src/corrgi/corrgi.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+from lsdb import Catalog
+from munch import Munch
+
+from corrgi.dask import compute_autocorrelation_counts
+from corrgi.estimators import calculate_natural_estimate
+
+
+def compute_autocorrelation(catalog: Catalog, random: Catalog, params: Munch) -> np.ndarray:
+    """Calculates the auto-correlation for a catalog.
+
+    Args:
+        catalog (Catalog): The catalog.
+        random (Catalog): A random samples catalog.
+        params (Munch): The parameters dictionary to run gundam with.
+
+    Returns:
+        A numpy array with the result of the auto-correlation, using the natural estimator.
+    """
+    num_galaxies = catalog.hc_structure.catalog_info.total_rows
```

Comment:
This works for now, but will not be a valid count if we perform some pre-filtering on the catalog:

```
filtered = read_hipscat(path, columns=[...], query =...).apply_more_filter()
result = compute_autocorrelation(filtered)
```

Reply:
Thank you Melissa, that's right! If there's filtering I should just be able to sum all the `counts_dd` and `counts_rr` and get the actual totals. I will open a small issue to address this.